### PR TITLE
fix(android): nested pager view 

### PR DESCRIPTION
**Pull request by NiuGuohui**
fix Issue (#303)

# Summary
See [here](https://github.com/android/views-widgets-samples/blob/master/ViewPager2/app/src/main/java/androidx/viewpager2/integration/testapp/NestedScrollableHost.kt)

I only tested example project. Looks okay.

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅    |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I updated the typed files (TS and Flow)

### CODE REVIEW

**Comment by NiuGuohui** [comment]
I also solved this in the Next branch.Do I need to open a new PR?

**Comment by troZee** [comment]
> I also solved this in the Next branch.Do I need to open a new PR?

Hey @NiuGuohui. I would like to check this PR, before you make an additional PR to the `next` branch.

**Comment by NiuGuohui** [comment]
> > I also solved this in the Next branch.Do I need to open a new PR?
> 
> Hey @NiuGuohui. I would like to check this PR, before you make an additional PR to the `next` branch.

ok

File: android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt [hunk 1]
```diff
@@ -85,68 +89,78 @@ class PagerViewViewManager : ViewGroupManager<ViewPager2>() {
...
 
   override fun needsCustomLayoutForChildren(): Boolean {
     return true
   }
 
   @ReactProp(name = "scrollEnabled", defaultBoolean = true)
-  fun setScrollEnabled(viewPager: ViewPager2, value: Boolean) {
-    viewPager.isUserInputEnabled = value
+  fun setScrollEnabled(host: NestedScrollableHost, value: Boolean) {
+    val view = getViewPager(host)
```
Comment (troZee) [inline android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt:136]: please merge those two lines into one line for each react prop

**Review by troZee — CHANGES_REQUESTED** [review]
Thank you @NiuGuohui for your investigation 🎉 
I have checked your changes and it works on my end. I added some comments to this PR. Fell free to start a discussion in case of any question.

**Comment by troZee** [comment]
@NiuGuohui Could you add additional example, which refers to nested pager in this PR ? Thank you in advance

File: android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt [hunk 2]
```diff
@@ -54,25 +55,28 @@ class PagerViewViewManager : ViewGroupManager<ViewPager2>() {
...
-      eventDispatcher.dispatchEvent(PageSelectedEvent(vp.id, vp.currentItem))
+      eventDispatcher.dispatchEvent(PageSelectedEvent(host.id, vp.currentItem))
     }
-
-    return vp
+    host.addView(vp)
+    return host
   }
 
+  private fun getViewPager(view: NestedScrollableHost) = view.getChildAt(0) as ViewPager2
```
[the remark this answers is not in the document]
Comment (NiuGuohui) [reply android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt:68]: It doesn't meet expectations if `childAt(0)` return type is not ViewPager2.
```
private fun getViewPager(view: NestedScrollableHost): ViewPager2 {
    if(view.getChildAt(0) is ViewPager2) {
      return view.getChildAt(0) as ViewPager2
    } else {
      throw ClassNotFoundException("Could not retrieve ViewPager2 instance")
    }
}
```

File: android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt [hunk 3]
```diff
@@ -17,14 +17,15 @@ import com.reactnativepagerview.event.PageScrollStateChangedEvent
...
   private lateinit var eventDispatcher: EventDispatcher
 
   override fun getName(): String {
     return REACT_CLASS
   }
 
-  override fun createViewInstance(reactContext: ThemedReactContext): ViewPager2 {
+  override fun createViewInstance(reactContext: ThemedReactContext): NestedScrollableHost {
+    val host = NestedScrollableHost(reactContext)
     val vp = ViewPager2(reactContext)
```
[the remark this answers is not in the document]
Comment (NiuGuohui) [reply android/src/main/java/com/reactnativepagerview/PagerViewViewManager.kt:29]: yeah!

**Comment by NiuGuohui** [comment]
> @NiuGuohui Could you add additional example, which refers to nested pager in this PR ? Thank you in advance

@troZee I added a nested example :)

File: android/src/main/java/com/reactnativepagerview/NestedScrollableHost.kt [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+ * (e.g. a horizontal RecyclerView in a vertical RecyclerView in a horizontal ViewPager2).
+ */
+class NestedScrollableHost : FrameLayout {
+  constructor(context: Context) : super(context)
+  constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+
+  private var touchSlop = 0
+  private var initialX = 0f
+  private var initialY = 0f
+  private val parentViewPager: ViewPager2?
```
[the remark this answers is not in the document]
Comment (troZee) [reply android/src/main/java/com/reactnativepagerview/NestedScrollableHost.kt:29]: @NiuGuohui how about this ?

**Comment by nathansbird** [comment]
@NiuGuohui This is super helpful, thanks! I can't seem to get it working on 0.63.3 (Latest version Expo supports). Any thoughts on why?

**Comment by NiuGuohui** [comment]
> @NiuGuohui This is super helpful, thanks! I can't seem to get it working on 0.63.3 (Latest version Expo supports). Any thoughts on why?

I don’t know Expo well, can it run in a 0.63.3 independent project?

**Comment by troZee** [comment]
> @NiuGuohui This is super helpful, thanks! I can't seem to get it working on 0.63.3 (Latest version Expo supports). Any thoughts on why?

Please check pager version in the expo SDK

**Comment by troZee** [comment]
https://github.com/callstack/react-native-pager-view/releases/tag/v5.4.3

**Pull request merged**